### PR TITLE
[datadog_rum_metric] Add support for the resource.

### PR DESCRIPTION
**Pull request by hemeryar**
Adds terraform support for RUM custom metrics. Leverages this public API:
https://docs.datadoghq.com/api/latest/rum-metrics/

https://datadoghq.atlassian.net/browse/RUM-5984

### CODE REVIEW

File: docs/resources/rum_metric.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `event_type` (String) The type of RUM events to filter on.
+- `name` (String) The name of the rum-based metric. This field can't be updated after creation.
```
Comment (cswatt) [inline docs/resources/rum_metric.md:45]:
```suggestion
- `name` (String) The name of the RUM-based metric. This field can't be updated after creation.
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: docs/resources/rum_metric.md (hunk 1 above)
Comment (hemeryar) [reply docs/resources/rum_metric.md:45]: Done

File: docs/resources/rum_metric.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+### Nested Schema for `compute`
+
+Required:
+
+- `aggregation_type` (String) The type of aggregation to use.
+
+Optional:
+
+- `include_percentiles` (Boolean) Toggle to include or exclude percentile aggregations for distribution metrics. Only present when `aggregation_type` is `distribution`.
+- `path` (String) The path to the value the rum-based metric will aggregate on. Only present when `aggregation_type` is `distribution`.
```
[the remark this answers is not in the document]
Comment (hemeryar) [reply docs/resources/rum_metric.md:68]: Done

File: docs/resources/rum_metric.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+<a id="nestedblock--filter"></a>
+### Nested Schema for `filter`
+
+Optional:
+
+- `query` (String) The search query - following the RUM search syntax.
```
[the remark this answers is not in the document]
Comment (hemeryar) [reply docs/resources/rum_metric.md:76]: Done

File: docs/resources/rum_metric.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+
+<a id="nestedblock--group_by"></a>
+### Nested Schema for `group_by`
+
+Optional:
+
+- `path` (String) The path to the value the rum-based metric will be aggregated over.
```
[the remark this answers is not in the document]
Comment (hemeryar) [reply docs/resources/rum_metric.md:84]: Done

File: docs/resources/rum_metric.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+- `tag_name` (String) Eventual name of the tag that gets created. By default, `path` is used as the tag name.
```
[the remark this answers is not in the document]
Comment (hemeryar) [reply docs/resources/rum_metric.md:85]: Done

**Review by cswatt — DISMISSED** [review]

**Review by amaskara-dd — DISMISSED** [review]

**Review by HantingZhang2 — APPROVED** [review]

**Pull request merged**